### PR TITLE
Addressing comments regarding where to build the query parameters forthe HTTP Client calls.

### PR DESCRIPTION
**Pull request by AndresRoblesMX**
# Description

Addressing comments regarding where to build the query parameters forthe HTTP Client calls.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by AndresRoblesMX** [comment]
@artursouza a small PR to improve the previous PR regarding the HTTP Client could you take a look?

File: sdk/src/main/java/io/dapr/client/domain/StateOptions.java [hunk 1]
```diff
@@ -29,6 +33,31 @@ public RetryPolicy getRetryPolicy() {
...
+        }
+        if (this.getRetryPolicy().getThreshold() != null) {
+          mapOptions.put("retryThreshold", this.getRetryPolicy().getThreshold().toString());
+        }
+        if (this.getRetryPolicy().getPattern() != null) {
+          mapOptions.put("retryPattern", this.getRetryPolicy().getPattern().getValue());
+        }
+      }
+    }
+    return mapOptions;
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/client/domain/StateOptions.java]: return Collections.unmodifiableMap(mapOptions);

File: sdk/src/main/java/io/dapr/client/DaprHttp.java [hunk 2]
```diff
@@ -151,9 +147,13 @@ public int getStatusCode() {
             } else {
               body = RequestBody.Companion.create(content, mediaType);
             }
+            HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+            urlBuilder.host(this.baseUrl).addPathSegment(urlString);
+            Optional.ofNullable(urlParameters).orElse(Collections.emptyMap()).entrySet().stream()
+                .forEach(urlParameter -> urlBuilder.addQueryParameter(urlParameter.getKey(), urlParameter.getValue()));
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/client/DaprHttp.java]: So much better, not?

File: sdk/src/main/java/io/dapr/client/domain/StateOptions.java (hunk 1 above)
Comment (AndresRoblesMX) [reply sdk/src/main/java/io/dapr/client/domain/StateOptions.java]: done, can you look at it again?

**Pull request merged**